src/calendarworker.cpp

```diff
@@ -759,12 +759,11 @@ CalendarWorker::dailyEventOccurrences(const QList<CalendarData::Range> &ranges,
         QDate start = range.first;
         while (start <= range.second) {
             foreach (const CalendarData::EventOccurrence &eo, occurrences) {
-                // On all day events the end time is inclusive, otherwise not
-                if ((eo.startTime.date() < start
-                     && (eo.endTime.date() > start
-                         || (eo.endTime.date() == start && (eo.eventAllDay
-                                                            || eo.endTime.time() > QTime(0, 0)))))
-                    || eo.startTime.date() == start) {
+                const QDateTime startDt(start);
+                // for exchange, the all-day event ends at 00:00 of the next day
```

Comment:
No it does not. Kcalendarcore sets the rules.

Reply:
done